### ISSUE TITLE
Minmal .NET API - Delegate 'RequestDelegate' does not take 0 arguments

### ISSUE THREAD

**Issue by verbedr** [dotnet/aspnetcore#47275]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Sort of like this https://github.com/dotnet/aspnetcore/issues/34390, but in my case I wanted to create stub routes to be completed later. These look like

`MapGet("countries", () => throw new NotImplementedException());`

### Expected Behavior

No warning/error

### Steps To Reproduce

Set up a minimal API like

`MapGet("countries", () =>throw new NotImplementedException());`

problem is due to the lack of return value.

### .NET Version

7.0.300-preview.23122.5; using VS2022 17.6.0 Preview 2.0

### Anything else?

Current solution is to write it as

`MapGet("countries", () => { throw new NotImplementedException(); return string.Empty; });`

This is by no means a blocking or problematic bug. I just want to report this behaviour from Visual Studio I encountered while migrating an old project to minimal API

**Comment by captainsafia** [comment]
@verbedr Yep, the issue here is that the delegate type cannot be inferred for the particular shape that you've provided. 

> MapGet("countries", () => { throw new NotImplementedException(); return string.Empty; });

Alternatively, `app.MapGet("/", () => { });` if you're not particular about needing to throw an excpetion.

**Comment by davidfowl** [comment]
This is also an option:

```C#
app.MapGet("countries", void () => throw new NotImplementedException());
```
